pilot/pkg/xds/ep_filters.go

```diff
@@ -64,6 +65,9 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocalityEndpoint
 		// Create a map to keep track of the gateways used and their aggregate weights.
 		gatewayWeights := make(map[model.NetworkGateway]uint32)
 
+		// Create a map to keep track of aggregate health of the gateways.
+		gatewayHealth := make(map[model.NetworkGateway]bool)
```

Comment:
nit use map[model.NetworkGateway]struct{} instead of map[model.NetworkGateway]bool

Reply:
Thanks, forgot about that option. Fixed.